controlplane/kubeadm/internal/cluster.go

```diff
@@ -86,7 +86,7 @@ func (m *Management) GetMachinesForCluster(ctx context.Context, cluster client.O
 func (m *Management) GetWorkloadCluster(ctx context.Context, clusterKey client.ObjectKey) (WorkloadCluster, error) {
 	// TODO(chuckha): Inject this dependency.
 	// TODO(chuckha): memoize this function. The workload client only exists as long as a reconciliation loop.
-	restConfig, err := remote.RESTConfig(ctx, m.Client, clusterKey)
+	restConfig, err := remote.RESTConfig(ctx, "kubeadm", m.Client, clusterKey)
```

Comment:
should this be "kubeadm-controlplane-controller"?

Reply:
yes that's better, updated. Thanks